src/main/fc/msp_server_fc.c

```diff
@@ -520,15 +541,21 @@ int mspServerCommandHandler(mspPacket_t *cmd, mspPacket_t *reply)
             sbufWriteU8(dst, FC_VERSION_PATCH_LEVEL);
             break;
 
-        case MSP_BOARD_INFO:
+        case MSP_BOARD_INFO: {
             sbufWriteData(dst, boardIdentifier, BOARD_IDENTIFIER_LENGTH);
 #ifdef USE_HARDWARE_REVISION_DETECTION
             sbufWriteU16(dst, hardwareRevision);
 #else
             sbufWriteU16(dst, 0); // No hardware revision available.
 #endif
-            sbufWriteU8(dst, 0);  // 0 == FC, 1 == OSD, 2 == FC with OSD
+#ifdef OSD
```

Comment:
no, we're free to change this.  and after writing the client side code I had some reservations with this approach but no good solution came to mind.  the OSD boards and FC boards do not share the same feature bitmask, so features cannot be used, hence board type.